### ISSUE TITLE
bug: Standalone Setup w/ Flakes instructions do not correctly describe how to track stable nixpkgs

### ISSUE THREAD

**Issue by matta** [nix-community/home-manager#5682]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

The instructions for a standalone installation with nix flakes are at https://nix-community.github.io/home-manager/index.xhtml#sec-flakes-standalone and say to use this command *"For Nixpkgs or NixOS version 24.05"*:

- `nix run home-manager/release-24.05 -- init --switch`

However, this creates a `flake.lock` file with the following contents:

```
{
  description = "Home Manager configuration of matt";

  inputs = {
    # Specify the source of Home Manager and Nixpkgs.
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
    home-manager = {
      url = "github:nix-community/home-manager";
      inputs.nixpkgs.follows = "nixpkgs";
    };
  };

  outputs = { nixpkgs, home-manager, ... }:
    let
      system = "x86_64-linux";
      pkgs = nixpkgs.legacyPackages.${system};
    in {
      homeConfigurations."matt" = home-manager.lib.homeManagerConfiguration {
        inherit pkgs;

        # Specify your home configuration modules here, for example,
        # the path to your home.nix.
        modules = [ ./home.nix ];

        # Optionally use extraSpecialArgs
        # to pass through arguments to home.nix
      };
    };
}
```

In other words, it produces the same `flake.nix` as if I had run `nix run home-manager/master -- init --switch`. After the switch things like `home-manager --version` do not print "24.05" but something like "24.11-pre".

To actually init home manager in a way that gets it using nixpkgs 24.05 I had to first undo the previous install:

```
$ nix profile remove home-manager-path
$ rm ~/.config/home-manager/flake.nix ~/.config/home-manager/flake.lock ~/.config/home-manager/home.nix
```

Then I installed home manager this way:

```
nix run home-manager/release-24.05 -- init \
    --switch \
    --nixpkgs-url github:nixos/nixpkgs/nixos-24.05 \
    --home-manager-url 'github:nix-community/home-manager/release-24.05'
```

Further down, the docs suggest that incantations like this should work (but they apparently won't in the general case):

```
nix run home-manager/$branch -- init
# Edit files in ~/.config/home-manager
nix run home-manager/$branch -- init --switch
```

I notice that the test for this feature passes these two flags explicitly, unlike the instructions above: https://github.com/nix-community/home-manager/blob/d0240a064db3987eb4d5204cf2400bc4452d9922/tests/integration/standalone/flake-basics.nix#L53

### Maintainer CC

@rycee, FYI your commit 33ccc5c28572d200f1e8d93fccba17e084e5f587 is when the current installation instructions for stable were introduced.

I'm not sure whether this is a bug in the docs or the implementation. It would be nice to not have to specify versions like 24.05 three times on the comnand line when installing home manager. That said, since home-manager aims to configure and install itself, there seems to be an installation time chicken-and-egg problem with the flake approach that doesn't happen with a channel based install.

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.1.0-23-amd64, Debian GNU/Linux, 12 (bookworm), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.23.3`
 - nixpkgs: `/nix/store/56m6nvsymx61kw3pkdibnwwz7r4y4cmb-source`
```

**Comment by DamienCassou** [comment]
I agree that instructions are confusing for a standalone flake-based install.

**Comment by kerrnelPanic** [comment]
I tried to do sth. like:

clone a repo with a flake
init home manager with that flake

I'm not sure if this is supposed to work, but with the instructions on the home manager install page and other information I could find online, I was not able to do this.

I tried to adapt the command on the instructions page:

`nix run home-manager/$branch -- init --switch ~/hmconf` 
->
`nix run home-manager/$branch -- init --switch ~/myFlakeDir#myConfig`

But it created a directory

`~/"myFlakeDir#myConfig"` 

instead and initialized a flake there. Omitting 

`#myConfig` 

left me with an error that indicates, hm tried to find a config in the flake at 

`myFlakeDir` 

that matches the name of the current user, which in my case didn't exist.
